functional-tests/fn_bridge_deposit_happy.py

```diff
@@ -44,114 +46,93 @@ def main(self, ctx: flexitest.RunContext):
         self.logger.debug(f"BTC user: {btc_user}")
         self.logger.debug(f"BTC password: {btc_password}")
 
+        # Get operators pubkey and musig2 aggregates it
+        bridge_pk = get_bridge_pubkey(seqrpc)
+        print(f"Bridge pubkey: {bridge_pk}")
+
         seq_addr = seq.get_prop("address")
         addr_1 = get_address(0)
         addr_2 = get_address(1)
-        self.logger.debug(f"EL address 1: {el_address_1}")
-        self.logger.debug(f"EL address 2: {el_address_2}")
-
-        original_balance_1 = int(rethrpc.eth_getBalance(f"0x{el_address_1}"), 16)
-        self.logger.debug(f"Balance before deposit (EL address 1): {original_balance_1}")
-
-        original_balance_2 = int(rethrpc.eth_getBalance(f"0x{el_address_2}"), 16)
-        self.logger.debug(f"Balance before deposit (EL address 2): {original_balance_2}")
+        self.logger.debug(f"Sequencer Address: {seq_addr}")
+        self.logger.debug(f"Address 1: {addr_1}")
+        self.logger.debug(f"Address 2: {addr_2}")
+
+        # Generate Plenty of BTC to address 1
+        original_balance = int(rethrpc.eth_getBalance(f"0x{el_address_1}"), 16)
+        self.logger.debug(f"Balance before deposit (EL address 1): {original_balance}")
+        btcrpc.proxy.generatetoaddress(102, addr_1)[0]
+        # Send DRT from Address 1 to EL Address 1
+        self.make_drt(ctx, el_address_1, bridge_pk)
+        new_balance = int(rethrpc.eth_getBalance(f"0x{el_address_1}"), 16)
+        self.logger.debug(f"Balance after deposit (EL address 1): {new_balance}")
+        assert new_balance > original_balance, "balance did not increase"
+        # Drain wallet back to sequencer so that we cannot use address 1 or change anymore
+        self.drain_wallet(ctx)
+
+        # Generate Plenty of BTC to address 2
+        original_balance = int(rethrpc.eth_getBalance(f"0x{el_address_2}"), 16)
+        self.logger.debug(f"Balance before deposit (EL address 2): {original_balance}")
+        btcrpc.proxy.generatetoaddress(102, addr_2)[0]
+        # Send DRT from Address 2 to EL Address 2
+        self.make_drt(ctx, el_address_2, bridge_pk)
+        new_balance = int(rethrpc.eth_getBalance(f"0x{el_address_2}"), 16)
+        self.logger.debug(f"Balance after deposit (EL address 2): {new_balance}")
+        assert new_balance > original_balance, "balance did not increase"
+        # Drain wallet back to sequencer so that we cannot use address 1 or change anymore
+        self.drain_wallet(ctx)
```

Comment:
Should refactor this to a separate function that does the checks. The following additional checks must be added:

1) Check the initial balance in the EL address
2) Assert that the difference in the balance after deposit is exactly equal to the `deposit_amount` (set in the rollup params).
3) Use the `getCurrentDeposits` RPC on the sequencer to check if the number of deposits increases as deposits are made.
4) Also check that multiple deposits to the same address works.

It's fine to split these tests out but it's preferable to not do that since setting up and tearing down takes a LOT of time and all these cases are part of the same "happy-case" environment.